app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt

```diff
@@ -61,18 +62,30 @@ class PatientsScreenControllerTest {
 
   private val uiEvents: PublishSubject<UiEvent> = PublishSubject.create()
   private lateinit var controller: PatientsScreenController
+  private val configEmitter = PublishSubject.create<PatientConfig>()
 
   @Before
   fun setUp() {
     // This is needed because we manually subscribe to the refresh user status
     // operation on the IO thread, which was causing flakiness in this test.
     RxJavaPlugins.setIoSchedulerHandler { Schedulers.trampoline() }
 
-    controller = PatientsScreenController(userSession, dataSync, patientRepository, appointmentRepository, approvalStatusApprovedAt, hasUserDismissedApprovedStatus, patientSummaryResult)
+    controller = PatientsScreenController(
+        userSession = userSession,
+        dataSync = dataSync,
+        patientRepository = patientRepository,
+        appointmentRepository = appointmentRepository,
+        approvalStatusUpdatedAtPref = approvalStatusApprovedAt,
+        hasUserDismissedApprovedStatusPref = hasUserDismissedApprovedStatus,
+        patientSummaryResult = patientSummaryResult,
+        configProvider = configEmitter
+    )
 
     uiEvents
         .compose(controller)
         .subscribe { uiChange -> uiChange(screen) }
+
+    configEmitter.onNext(PatientConfig(limitOfSearchResults = 1, scanSimpleCardFeatureEnabled = false))
```

Comment:
Not important, but I've started keeping a template config object around which test functions can use.

```kotlin
class SomeTest {
  
  // All default values.
  private val configTemplate = PatientConfig(limitOfSearchResults = 1, scanSimpleCardFeatureEnabled = false)

  @Test
  fun foo() {
    configEmitter.onNext(configTemplate.copy(limitOfSearchResults = 2)
  }
}
```

Advantages:
- Tests can focus on changing values only they care about
- As the number of config fields grow over time, you won't have to update every test

Reply:
Good point. I'll change this when I need to modify this test class next.